[2024-05-19T11:08:00.878+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-05-19T11:08:00.929+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: dummyjson.transform_cart manual__2024-05-19T11:07:31.553907+00:00 [queued]>
[2024-05-19T11:08:00.958+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: dummyjson.transform_cart manual__2024-05-19T11:07:31.553907+00:00 [queued]>
[2024-05-19T11:08:00.960+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 6
[2024-05-19T11:08:01.000+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): transform_cart> on 2024-05-19 11:07:31.553907+00:00
[2024-05-19T11:08:01.015+0000] {warnings.py:110} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61: DeprecationWarning: This process (pid=2292) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2024-05-19T11:08:01.020+0000] {standard_task_runner.py:63} INFO - Started process 2299 to run task
[2024-05-19T11:08:01.020+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'dummyjson', 'transform_cart', 'manual__2024-05-19T11:07:31.553907+00:00', '--job-id', '1408', '--raw', '--subdir', 'DAGS_FOLDER/dummyjson.py', '--cfg-path', '/tmp/tmpqc_tkm8v']
[2024-05-19T11:08:01.027+0000] {standard_task_runner.py:91} INFO - Job 1408: Subtask transform_cart
[2024-05-19T11:08:01.132+0000] {task_command.py:426} INFO - Running <TaskInstance: dummyjson.transform_cart manual__2024-05-19T11:07:31.553907+00:00 [running]> on host ca927d78a9ea
[2024-05-19T11:08:01.750+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_EMAIL='fathurrahmanhernanda@gmail.com' AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='dummyjson' AIRFLOW_CTX_TASK_ID='transform_cart' AIRFLOW_CTX_EXECUTION_DATE='2024-05-19T11:07:31.553907+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-05-19T11:07:31.553907+00:00'
[2024-05-19T11:08:01.752+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-05-19T11:08:01.777+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-05-19T11:08:01.778+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 400, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/dummyjson.py", line 96, in transform_generic
    response = json.loads(ti.xcom_pull(task_ids=f"extract_{kwargs['data_type']}"))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/json/__init__.py", line 339, in loads
    raise TypeError(f'the JSON object must be str, bytes or bytearray, '
TypeError: the JSON object must be str, bytes or bytearray, not NoneType
[2024-05-19T11:08:01.802+0000] {taskinstance.py:1206} INFO - Marking task as UP_FOR_RETRY. dag_id=dummyjson, task_id=transform_cart, run_id=manual__2024-05-19T11:07:31.553907+00:00, execution_date=20240519T110731, start_date=20240519T110800, end_date=20240519T110801
[2024-05-19T11:08:01.828+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 1408 for task transform_cart (the JSON object must be str, bytes or bytearray, not NoneType; 2299)
[2024-05-19T11:08:01.890+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2024-05-19T11:08:01.963+0000] {taskinstance.py:3498} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-05-19T11:08:01.967+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
